doc/radosgw/s3select.rst

```diff
@@ -0,0 +1,265 @@
+
+    | **Alias** programming-construct is an essential part of s3-select language, it enables much better programming especially with objects containing many columns or in the case of complex queries.
+    
+    | Upon parsing the statement containing alias construct, it replaces alias with reference to the correct AST-node, on runtime the node is simply evaluated as any other node.
+
+    | There is a risk that self(or cyclic) reference may occur causing stack-overflow(endless-loop), for that concern upon evaluating an alias, it is validated for cyclic reference.
+    
+    | Alias also maintains result-cache, meaning upon using the same alias more than once, it’s not evaluating the same node again(it will return the same result),instead it uses the result from cache.
+
+    | Of Course, per each new row the cache is invalidated.
+        
+
+S3 select parser definition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    | The implementation of s3-select uses the `boost::spirit <https://www.boost.org/doc/libs/1_71_0/libs/spirit/classic/doc/grammar.html>`_ the definition of s3-select command is according to AWS.
+     
+    | Upon parsing is initiated on input text, and a specific rule is identified, an action which is bound to that rule is executed.
+    | Those actions are building the AST, each action is unique (as its rule), at the end of the process it forms a structure similar to a tree. 
+    
+    | As mentioned, running eval() on the root node, execute the s3-select statement (per projection).
+    | The input stream is accessible to the execution tree, by the scratch-area object, that object is constantly updated per each new row. 
+
+Basic functionalities
+~~~~~~~~~~~~~~~~~~~~~
+
+    | **S3select** has a definite set of functionalities that should be implemented (if we wish to stay compliant with AWS), currently only a portion of it is implemented.
+    
+    | The implemented software architecture supports basic arithmetic expressions, logical and compare expressions, including nested function calls and casting operators, that alone enables the user reasonable flexibility. 
+    | review the bellow feature-table_.
+
+
+
+Memory handling
+~~~~~~~~~~~~~~~
+
+    | S3select structures and objects are lockless and thread-safe, it uses placement-new in order to reduce the alloc/dealloc intensive cycles, which may impact the main process hosting s3-select.
+    
+    | Once AST is built there is no need to allocate memory for the execution itself, the AST is “static” for the query-execution life-cycle.
+    
+    | The execution itself is stream-oriented, meaning there is no pre-allocation before execution, object size has no impact on memory consumption.
+    
+    | It processes chunk after chunk, row after row, all memory needed for processing resides on AST. 
+    
+    | The AST is similar to stack behaviour in that it consumes already allocated memory and “releases” it upon completing its task.
+
+S3 Object different types
```

Comment:
this section should only include detailed definition of the CSV format we support

Reply:
what does it mean "detailed definitions" ? there is section with examples.

Reply:
expect from example, give gitails on how we behave regarding quotes, escapes, newlines etc.
if we comply to some standard you can give reference to it, and just highlight differences